src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -393,27 +394,125 @@ private Task StartAsyncCore(TransferFormat transferFormat)
             }
         }
 
-        private static async Task<NegotiationResponse> ParseNegotiateResponse(HttpResponseMessage response, ILogger logger)
+        private static async Task<NegotiationResponse> ParseNegotiateResponse(HttpResponseMessage response)
```

Comment:
Move parsing the negotiate response out of HttpConnection? If yes, should the new class also handle writing a negotiate response?

Reply:
Yes, please. Lets do this please. This looks messy inline.

Reply:
Also do you expect that this will be generally faster?

Reply:
Absolutely. There is lots of overhead in serializing: allocating serializer, internal serializer state, getting serialization contract, testing various settings, etc. Want benchmarks?

Reply:
Yea, benchmark this thingy.